src/main/java/com/epam/esm/controller/TagsController.java

```diff
@@ -33,36 +35,45 @@ public TagsController(TagService service) {
     /**
      * Returns an {@link ResponseEntity} object contained {@link HttpStatus} status and a {@link List} list of {@link Tag}
      * tags retrieved from database.
+     * All parameters are optional.
      *
+     * @param page     specifies page of the result list.
+     * @param pageSize specifies number of tags to be displayed in a single page. In case page were passed without page size
+     *                   default page size applies.
      * @return {@link ResponseEntity} contained both {@link HttpStatus} status and {@link List} of {@link Tag} tags.
      */
     @GetMapping(produces = JSON)
-    public ResponseEntity<List<Tag>> getTags() {
-        return new ResponseEntity<>(service.getTags(), HttpStatus.OK);
+    public ResponseEntity<CollectionModel<Tag>> getTags(@RequestParam(required = false) Integer page,
+                                                        @RequestParam(required = false) Integer pageSize) {
+        List<Tag> tags = service.getTags(page, pageSize);
+        tags.forEach(tag -> tag.add(linkTo(methodOn(TagsController.class).getTag(tag.getId())).withSelfRel()));
+        Link allTagsLink = linkTo(methodOn(TagsController.class).getTags(page, pageSize)).withSelfRel();
```

Comment:
It would be better to add link to the next and previous pages

Reply:
Added